plasmapy/diagnostics/thomson.py

```diff
@@ -204,114 +444,372 @@ def spectral_density(
+        nums = ["ifract_" + str(i) for i in range(num_i - 1)]
+        nums.insert(0, "1.0")
+        params["ifract_" + str(num_i - 1)].expr = " - ".join(nums)
+
+    # **************
+    # Electron velocity
+    # **************
+    electron_speed = np.zeros([num_e])
+    for e in range(num_e):
+        k = "electron_speed_" + str(e)
+        if k in list(params.keys()):
+            electron_speed[e] = params[k].value
+        else:
+            # electron_speed[e] = 0 already
+            params.add(k, value=0, vary=False)
+
+    if "electron_vdir" not in list(settings.keys()):
+        if np.all(electron_speed == 0):
+            # vdir is arbitrary in this case because vel is zero
+            settings["electron_vdir"] = np.ones([num_e, 3])
+        else:
+            raise ValueError(
+                "electron_vdir must be set if electron_speeds " "are not all zero."
+            )
+    # Normalize vdir
+    norm = np.linalg.norm(settings["electron_vdir"], axis=-1)
+    settings["electron_vdir"] = settings["electron_vdir"] / norm[:, np.newaxis]
+
+    # **************
+    # Ion velocity
+    # **************
+    ion_speed = np.zeros([num_i])
+    for i in range(num_i):
+        k = "ion_speed_" + str(i)
+        if k in list(params.keys()):
+            ion_speed[i] = params[k].value
+        else:
+            # ion_speed[i] = 0 already
+            params.add(k, value=0, vary=False)
+
+    if "ion_vdir" not in list(settings.keys()):
+        if np.all(ion_speed == 0):
+            # vdir is arbitrary in this case because vel is zero
+            settings["ion_vdir"] = np.ones([num_i, 3])
+        else:
+            raise ValueError("ion_vdir must be set if ion_speeds " "are not all zero.")
+    # Normalize vdir
+    norm = np.linalg.norm(settings["ion_vdir"], axis=-1)
+    settings["ion_vdir"] = settings["ion_vdir"] / norm[:, np.newaxis]
+
+    if "inst_fcn" not in list(settings.keys()):
+        settings["inst_fcn_arr"] = None
+    else:
+        # Create inst fcn array from inst_fcn
+        inst_fcn = settings["inst_fcn"]
+        wspan = (np.max(wavelengths) - np.min(wavelengths)) / 2
+        eval_w = np.linspace(-wspan, wspan, num=wavelengths.size)
+        inst_fcn_arr = inst_fcn(eval_w * u.m)
+        inst_fcn_arr *= 1 / np.sum(inst_fcn_arr)
+        settings["inst_fcn_arr"] = inst_fcn_arr
+
+    # TODO: raise an exception if the number of any of the ion or electron
+    # quantities isn't consistent with the number of that species defined
+    # by ifract or efract.
+
+    # Create a lmfit.Model
+    # nan_policy='omit' automatically ignores NaN values in data, allowing those
+    # to be used to represnt regions of missing data
+    # the "settings" dict is an additional kwarg that will be passed to the model function on every call
+    model = Model(
+        _spectral_density_model,
+        independent_vars=["wavelengths"],
+        nan_policy="omit",
+        settings=settings,
+    )
+
+    return model
```

Comment:
```suggestion
    # Create and return the lmfit.Model
    return Model(
        _spectral_density_model,
        independent_vars=["wavelengths"],
        nan_policy="omit",
        settings=settings,
    )
```

The comment can be reduced since the two points it makes are "easily" recognizable from reading the code block. If the code block used `return lmfit.Model(...)` instead of just `return Model(...)`, then I would have suggested to remove the comment altogether.